### PR TITLE
Addition of data quality control and fault detection app and documentation

### PR DESCRIPTION
**Pull request by kaklise**
## Summary/Motivation:
This PR leverages functionality in Pecos (https://github.com/sandialabs/pecos) to add data quality control and fault detection analysis to IDAES.  

## Changes proposed in this PR:
- Pecos is a Python package that can be used with IDAES for data quality control and fault detection analysis
- The basic functionality is documented in the user manual under a new section titled "Related Packages".  
- A separate PR will be submitted to examples-pse to demonstrate use cases and test functionality.

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: docs/related_packages/pecos.rst [hunk 1]
```diff
@@ -0,0 +1,68 @@
+.. _pecos:
+
+Pecos: Data Quality Control and Fault Detection
+=================================================
+
+Before using plant data in process models, quality control and fault detection analysis is recommended to identify 
+potential data issues (i.e., missing or corrupt data) and data points that are not suitable for the intended analysis 
+(i.e., abnormal plant behavior).
```
Comment (eslickj) [inline docs/related_packages/pecos.rst:8]: Super picky, but did you mean e.g. here (in place of i.e.)?

File: docs/related_packages/pecos.rst [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+This following documentation describes methods to run data quality control and fault detection analysis using Pecos.
```
Comment (eslickj) [inline docs/related_packages/pecos.rst:9]: The following...

or 

This documentation...

File: docs/related_packages/pecos.rst [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+To install Pecos using git::
+
+	git clone https://github.com/sandialabs/pecos
+	cd pecos
+	python setup.py install
+	
+Examples
+----------
+IDAES examples that use Pecos are listed on the |examples-site|. 
+Pecos also includes several general examples, located at https://github.com/sandialabs/pecos/tree/master/examples
```
Comment (eslickj) [inline docs/related_packages/pecos.rst:68]: Maybe I missed it, but probably need a link to the pecos docs if it's not there.  I saw it in the example notebook, but I don't think I saw it here.

**Review by eslickj — DISMISSED** [review]
Looks good.  I have a few minor comments, mostly just to prove I read it.

File: docs/related_packages/index.rst [hunk 4]
```diff
@@ -0,0 +1,10 @@
+﻿Related Packages
+==================
+
+In addition to the IDAES PSE framework, several related software packages have been developed and/or
```
[the remark this answers is not in the document]
Comment (kaklise) [reply docs/related_packages/index.rst:4]: Yes, there will be more packages that fall into this category.   My understanding is that these will be added soon.

File: docs/related_packages/pecos.rst (hunk 1 above)
Comment (kaklise) [reply docs/related_packages/pecos.rst]: done

File: docs/related_packages/pecos.rst (hunk 2 above)
Comment (kaklise) [reply docs/related_packages/pecos.rst]: Done

File: docs/related_packages/pecos.rst (hunk 3 above)
Comment (kaklise) [reply docs/related_packages/pecos.rst]: The rst file now includes a link to Pecos docs and the repo

**Review by eslickj — DISMISSED** [review]
Looks good.

File: docs/related_packages/pecos.rst [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+* Cleaned data (data that failed a test are removed)
+* Boolean mask (indicates which data points failed a test)
+* Summary of the quality control test results (includes the variable name, start and end time for each failure, and an error message)
+
+The test results summary and accompanying graphics can be included in HTML or LATEX reports generated using Pecos.
+
+Pecos supports both static and streaming analysis along with custom quality control functions:
+ 
+* Static analysis operates on the entire data set to determine if all data points are normal or anomalous. 
+  While this can include operations like moving window statistics, the quality control tests operates on the entire data set at once.
```
Comment (blnicho) [inline docs/related_packages/pecos.rst:41]: Typo: "operates" should be "operate"

File: docs/related_packages/pecos.rst [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+* Boolean mask (indicates which data points failed a test)
+* Summary of the quality control test results (includes the variable name, start and end time for each failure, and an error message)
+
+The test results summary and accompanying graphics can be included in HTML or LATEX reports generated using Pecos.
+
+Pecos supports both static and streaming analysis along with custom quality control functions:
+ 
+* Static analysis operates on the entire data set to determine if all data points are normal or anomalous. 
+  While this can include operations like moving window statistics, the quality control tests operates on the entire data set at once. 
+* Streaming analysis loops through each data point using a quality control tests that relies on information from "clean data" in a moving window.
```
Comment (blnicho) [inline docs/related_packages/pecos.rst:42]: Typo: "tests" should be "test"

File: docs/related_packages/pecos.rst [hunk 7]
```diff
@@ -0,0 +1,70 @@
+.. _pecos:
+
+Pecos: Data Quality Control and Fault Detection
+=================================================
```
Comment (blnicho) [inline docs/related_packages/pecos.rst:4]: rst formatting, heading underline is longer than the text

File: docs/related_packages/pecos.rst [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+Data points that do not pass quality control inspection should be
+removed or replaced by various means (interpolation, data from a duplicate sensor, values from a model) before using the data for further analysis.
+Data replacement strategies are generally defined on a case-by-case basis. 
+If large sections of the data failed quality control tests, the data might not be suitable for use.
+
+The raw data, results from the quality control analysis, and the analysis files used to run Pecos can be stored in the 
+Data Management Framework (DMF) to ensure reproducible results.
+
+Installation
+---------------
```
Comment (blnicho) [inline docs/related_packages/pecos.rst:55]: rst formatting, heading underline is longer than the text

File: docs/related_packages/pecos.rst [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+	pip install pecos 
+	
+To install Pecos using git::
+
+	git clone https://github.com/sandialabs/pecos
+	cd pecos
+	python setup.py install
+	
+Examples
+----------
```
Comment (blnicho) [inline docs/related_packages/pecos.rst:68]: rst formatting, heading underline is longer than the text

File: docs/related_packages/index.rst [hunk 10]
```diff
@@ -0,0 +1,10 @@
+﻿Related Packages
+==================
```
Comment (blnicho) [inline docs/related_packages/index.rst:2]: rst formatting, heading underline is longer than the text

**Review by blnicho — COMMENTED** [review]
I found a few minor things but otherwise this looks great

File: docs/related_packages/pecos.rst (hunk 5 above)
Comment (kaklise) [reply docs/related_packages/pecos.rst]: done

File: docs/related_packages/pecos.rst (hunk 6 above)
Comment (kaklise) [reply docs/related_packages/pecos.rst]: done

File: docs/related_packages/pecos.rst (hunk 8 above)
Comment (kaklise) [reply docs/related_packages/pecos.rst]: done

File: docs/related_packages/pecos.rst (hunk 9 above)
Comment (kaklise) [reply docs/related_packages/pecos.rst]: done

File: docs/related_packages/index.rst (hunk 10 above)
Comment (kaklise) [reply docs/related_packages/index.rst]: done

**Review by eslickj — APPROVED** [review]

**Review by jsiirola — APPROVED** [review]

**Review by blnicho — APPROVED** [review]

**Pull request merged**